test/studies/hpcc/STREAMS/elliot/stream-spmd-barrier.chpl

```diff
@@ -35,9 +35,8 @@ proc main() {
 
   var execTime: [1..numTrials] real;
 
+  var barrier = new Barrier(numTasks, barrierType);
```

Comment:
Doesn't this need to be `reset` between trials?

Reply:
I don't think so, since it's a reusable barrier.

Reply:
Can the `halt` at `Barriers.chpl:358` ever be reached?  The fact that it doesn't need to be `reset` seems to imply that any number of tasks can call into the barrier.

Reply:
It can, but it's racey (as is the aBarrier version)


```chpl
var b = new Barrier(here.maxTaskPar);
coforall 1..here.maxTaskPar+1 do
  b.barrier();
```

will halt if all tasks hit the barrier before any of them get released from it.

I was going to add a test for this, but that's when I realized it's racey because it only failed every other time or so.